dm/relay/streamer.go

```diff
@@ -57,8 +59,18 @@ func (s *LocalStreamer) GetEvent(ctx context.Context) (*replication.BinlogEvent,
 		heartbeatInterval = time.Duration(i) * time.Second
 	})
 
+	fired := false
+	s.heatBeatTimer.Reset(heartbeatInterval)
+	defer func() {
```

Comment:
if `fired` we should also clean the tick in `s.heatBeatTimer.C`, in case the next `GetEvent` is called very slowly (maybe the heartbeat triggers a flushing checkpoint and cost more than 30s 😂 ) and read a staled tick.

Reply:
if fired, s.heatBeatTimer.C is already drained.